application/src/test/java/com/sanction/thunder/email/EmailModuleTest.java

```diff
@@ -14,81 +14,171 @@
     when(EMAIL_CONFIG.getRegion()).thenReturn("us-east-1");
     when(EMAIL_CONFIG.getFromAddress()).thenReturn("test@test.com");
+
+    String customSuccessHtmlFilePath = new File(
+        Resources.getResource(CUSTOM_SUCCESS_HTML_RESOURCE_FILE).toURI()).getAbsolutePath();
+    String customBodyHtmlFilePath = new File(
+        Resources.getResource(CUSTOM_BODY_HTML_RESOURCE_FILE).toURI()).getAbsolutePath();
+    String customBodyTextFilePath = new File(
+        Resources.getResource(CUSTOM_BODY_TEXT_RESOURCE_FILE).toURI()).getAbsolutePath();
+
+    when(OPTIONS_CONFIG.getBodyHtmlFilePath()).thenReturn(customBodyHtmlFilePath);
+    when(OPTIONS_CONFIG.getBodyTextFilePath()).thenReturn(customBodyTextFilePath);
+    when(OPTIONS_CONFIG.getSuccessHtmlFilePath()).thenReturn(customSuccessHtmlFilePath);
+  }
+
+  /* provideMessageOptions() */
+
+  @Test
+  void testProvideMessageOptionsNull() {
+    when(EMAIL_CONFIG.getMessageOptionsConfiguration()).thenReturn(null);
+
+    MessageOptions expected = new MessageOptions(
+        "Account Verification", "bodyHtml", "bodyText", "CODEGEN-URL", "successHtml");
+
+    MessageOptions result = new EmailModule(EMAIL_CONFIG)
+        .provideMessageOptions("bodyHtml", "bodyText", "successHtml");
+
+    assertEquals(expected, result);
+  }
+
+  @Test
+  void testProvideMessageOptionsCustom() {
+    when(OPTIONS_CONFIG.getSubject()).thenReturn("Test Subject");
+    when(OPTIONS_CONFIG.getUrlPlaceholderString()).thenReturn("Test Placeholder");
+    when(EMAIL_CONFIG.getMessageOptionsConfiguration()).thenReturn(OPTIONS_CONFIG);
+
+    MessageOptions expected = new MessageOptions(
+        "Test Subject", "bodyHtml", "bodyText", "Test Placeholder", "successHtml");
+
+    MessageOptions result = new EmailModule(EMAIL_CONFIG)
+        .provideMessageOptions("bodyHtml", "bodyText", "successHtml");
+
+    assertEquals(expected, result);
+  }
+
+  /* provideSuccessHtml() */
+
+  @Test
+  void testProvideSuccessHtmlNullOptions() throws IOException {
+    when(EMAIL_CONFIG.getMessageOptionsConfiguration()).thenReturn(null);
+
+    EmailModule emailModule = new EmailModule(EMAIL_CONFIG);
+
+    String expected = Resources.toString(
+        Resources.getResource(DEFAULT_SUCCESS_HTML_RESOURCE_FILE), Charsets.UTF_8);
+    assertEquals(expected, emailModule.provideSuccessHtml());
   }
 
   @Test
   void testProvideSuccessHtmlDefault() throws IOException {
-    when(EMAIL_CONFIG.getSuccessHtmlPath()).thenReturn(null);
+    when(EMAIL_CONFIG.getMessageOptionsConfiguration())
+        .thenReturn(new MessageOptionsConfiguration());
 
     EmailModule emailModule = new EmailModule(EMAIL_CONFIG);
 
     String expected = Resources.toString(
-        Resources.getResource("success.html"), Charsets.UTF_8);
+        Resources.getResource(DEFAULT_SUCCESS_HTML_RESOURCE_FILE), Charsets.UTF_8);
     assertEquals(expected, emailModule.provideSuccessHtml());
   }
 
   @Test
   void testProvideSuccessHtmlCustom() throws Exception {
```

Comment:
same here, just make sure the file uses it. I think assertThrows() is much cleaner!